File: C#/WP7/source/LFX-0.3.3/Loon.Core.Resource/LPKTable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Loon.Core.Resource
{
    public class LPKTable
    {

        private byte[] fileName;

        private long fileSize;

        private long offSet;

        public LPKTable()
        {
            this.fileSize = 0x0L;
            this.offSet = 0x0L;
            this.fileName = new byte[LPKHeader.LF_FILE_LENGTH];
        }

        public LPKTable(byte[] fileName_0, long s1, long s2)
        {
            this.fileSize = 0x0L;
            this.offSet = 0x0L;
            this.fileName = new byte[LPKHeader.LF_FILE_LENGTH];
            for (int i = 0; i < LPKHeader.LF_FILE_LENGTH; this.fileName[i] = fileName_0[i], i++)
                ;
            this.fileSize = s1;
            this.offSet = s2;
        }

        public byte[] GetFileName()
        {
            return fileName;
        }

        public void SetFileName(byte[] b)
        {
            for (int i = 0; i < b.Length; this.fileName[i] = b[i], i++)
                ;
        }

        public long GetFileSize()
        {
            return fileSize;
        }

        public void SetFileSize(long s)
        {
            this.fileSize = s;
        }

        public long GetOffSet()
        {
            return offSet;
        }

        public void SetOffSet(long s)
        {
            this.offSet = s;
        }

        public static int Size()
        {
            return LPKHeader.LF_FILE_LENGTH + 4 + 4;
        }
    }
}

File: Java/JavaSE/OpenGL/source/core/src/loon/core/resource/LPKTable.java
package loon.core.resource;

import java.io.Serializable;

/**
 * Copyright 2008 - 2009
 * 
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not
 * use this file except in compliance with the License. You may obtain a copy of
 * the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
 * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
 * License for the specific language governing permissions and limitations under
 * the License.
 * 
 * @project loon
 * @author cping
 * @email：javachenpeng@yahoo.com
 * @version 0.1
 */
public class LPKTable implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	private String fileName;

	private long fileSize = 0L;

	private long offSet = 0L;

	public LPKTable() {
		this.fileName = null;
	}

	public LPKTable(byte[] fileName, long fileSize, long offSet) {
		this.fileName = new String(fileName).trim();
		this.fileSize = fileSize;
		this.offSet = offSet;
	}

	public String getFileName() {
		return fileName;
	}

	public void setFileName(byte[] bytes) {
		this.fileName = new String(bytes).trim();
	}

	public long getFileSize() {
		return fileSize;
	}

	public void setFileSize(long fileSize) {
		this.fileSize = fileSize;
	}

	public long getOffSet() {
		return offSet;
	}

	public void setOffSet(long offSet) {
		this.offSet = offSet;
	}

	public static int size() {
		return LPKHeader.LF_FILE_LENGTH + 4 + 4;
	}

}
